odoo/addons/base/models/qweb.py

```diff
@@ -840,822 +629,545 @@ def _compile_static_node(self, el, options):
-                        ],
-                        keywords=[], starargs=None, kwargs=None
-                    )
-                ),
-                # if default_content:
-                #    display the tag with default content
-                # elif force_display:
-                #    display the tag without content
-                ast.If(
-                    test=ast.Name(id=default_content, ctx=ast.Load()),
-                    body=self._compile_tag(el, [self._append(ast.Name(id=default_content, ctx=ast.Load()))], options, True) or [ast.Pass()],
-                    orelse=orelse,
-                )
-            ]
-
-        # if content is not None:
-        #    display the tag (to_text(content))
-        # else
-        #    if default_content:
-        #       display the tag with default content
-        #    elif force_display:
-        #       display the tag without content
-        return [self._if_content_is_not_Falsy(
-            body=self._compile_tag(el, [self._append(
-                ast.Call(
-                    func=ast.Name(id='to_text', ctx=ast.Load()),
-                    args=[ast.Name(id='content', ctx=ast.Load())], keywords=[],
-                    starargs=None, kwargs=None
-                )
-            )], options, True),
-            orelse=orelse,
-        )]
-
-    def _compile_directive_call(self, el, options):
-        tmpl = el.attrib.pop('t-call')
-        _values = self._make_name('values_copy')
-        call_options = el.attrib.pop('t-call-options', None)
-        nsmap = options.get('nsmap')
+        code = []
+        code_options = self._compile_options(el, 't_field_t_options', options, indent)
+        if code_options:
+            code.extend(code_options)
+        else:
+            code.append(self._indent('t_field_t_options = {}', indent))
+
+        code.extend(self._compile_hook_before_directive(el, directive, options, indent))
+        code.append(self._indent(f"""attrs, content, force_display = self._get_field({self._compile_expr(record, raise_on_missing=True)}, "{self._compile_str(field_name)}", "{self._compile_str(expression)}", "{self._compile_str(tagName)}", t_field_t_options, compile_options, values)""", indent))
+        code.extend(self._compile_widget_value(el, directive, options, indent))
+        return code
 
+    def _compile_widget_value(self, el, directive, options, indent=0, without_attributes=False):
+        el.attrib.pop('t-tag', None)
+
+        code = self._flushText(options, indent)
+        code.append(self._indent("""if content is not None and content is not False:""", indent))
+        hook_after = self._compile_hook_after_directive(el, directive, options, indent + 1)
+        code.extend(hook_after)
+        code.extend(self._compile_tag_open(el, options, indent + 1, not without_attributes))
+        code.extend(self._flushText(options, indent + 1))
+        code.append(self._indent(dedent("""
+                if isinstance(content, (GeneratorType, list)):
```

Comment:
Why just these rather than, say, `isinstance(content, Iterator) and not isinstance(content, str)`?

Reply:
Not all values were strings. I found the source. Some widget returns as html value in int. This being fixed at the source I can indeed simply check if it is str.

Reply:
I think you misunderstood what I wrote. The code checked if the content was specifically a `Generator` or a `list`, which are but two specific cases of iterables. I suggested using a more general type (the wrong one tho, should have been `Iterable`) however special case is needed to *exclude strings* from being treated as iterables here: technically they are iterables but we want to treat them as atoms (non-iterables).

An `int` is never iterable, so it would always use the "atom" path.